### DISCUSSION TITLE
Update a record and also return other items in a single call

### DISCUSSION THREAD

**Discussion by Chris-Mingay** [supabase/supabase#4175] [Questions]
Hi all, I am just starting out with Supabase and am looking to create a quiz app using svelte as the frontend.

Let's say the question table has three properties id, category and question_text
I also have another table called question_options which has three properties id, question_id and option_text

To start with the user will only see the category and they will have an option to reveal the multiple choice question. At this point they get thirty seconds to answer the question presented to them.

So in svelte, I have a `revealQuestion()` function which I would like to do the following things:
- Log with supabase when the user requested the question details (so they can't cheat basically!)
- Return the question text and also return the question options

So this is a mixture of an `insert` and a `select` however the select returns two sets of data, the question text and an array of options. e.g.

```
{
  "question_text":"How bout them apples?",
  "question_options": [
    "They great",
    "Yeah not bad",
    "What's an apple?"
  ]
}
```

I believe I can do something like this with postgres functions however it seems like I am kind of fudging the functionality and missing the point of using supabase at all. Maybe it's not suited for this kind of application? That's a genuine question and not to sound dismissive at all, I honestly don't know enough about supabase yet to know for sure.

The alternative is to have a series of individual async calls from svelte, one to log the user request, another to get the question text and the final one to get the question options. But I guess by that point it doesn't really need to be supabase.

So in summary it feels like I am falling over because I don't really know how best to implement "backend" logic, am I missing something?

Thank you!

**Comment by steve-chavez** [comment]
> Update a record and also return other items in a single call 

The above is supported: https://postgrest.org/en/v9.0/api.html#embedding-after-insertions-updates-deletions

Maybe try it like:

```js
const { data, error } = await supabase
  .from('cities')
  .update({ name: 'Middle Earth' })
  .select(`
    name,
    countries (
      name
    )
  `)
```

**Comment by Chris-Mingay** [comment]
Thanks @steve-chavez unless I'm mistaken that's putting the responsibility of updating the record on the frontend? That's something I'd like to avoid because as far as I see it, I am not in control of how the frontend app behaves as soon as it's downloaded by a user.

**Reply by bohan0** [reply]
You are supposed to use Row Level Security policies to restrict what the user can update. Its a different way of thinking about security (vs traditional method of making server side functions to restrict what the user can update)

**Reply by Chris-Mingay** [reply]
Thanks for the response @bohan0 in this instance I want to guarantee that the update actually occurs when a specific record is requested, I don't want to rely on the front end code to instigate this update (i.e. the user if they wanted could update the code to remove the update() section of the supabase request.)
